singer/src/main/java/com/pinterest/singer/writer/KafkaWriter.java

```diff
@@ -212,6 +252,10 @@ private void includeAuditMessageInBatch(List<ProducerRecord<byte[], byte[]>> mes
           partitionId = validPartitions.get(partitionId).partition();
         }
         keyedMessage = new ProducerRecord<>(topic, partitionId, key, msg.getMessage());
+        if (enableHeadersProvider && SingerSettings.getHeadersProvider() != null){
```

Comment:
Is it possible to & enableHeadersProvider with `SingerSettings.getHeadersProvider() != null` in constructor to save 1 condition evaluation?

Reply:
yes. it is possible. I will create LoggingAuditHeaderProvider instance per Writer to prevent the above issues and do the null check it in the constructor